ksqldb-engine/src/main/java/io/confluent/ksql/util/QueryMetadata.java

```diff
@@ -339,6 +362,38 @@ private void doClose(final boolean cleanUp) {
     onStop.accept(cleanUp);
   }
 
+  public static class TimeBoundedQueue {
+    private final Duration duration;
+    private final Queue<QueryError> queue;
+
+    TimeBoundedQueue(final Duration duration, final int capacity) {
+      queue = EvictingQueue.create(capacity);
+      this.duration = duration;
+    }
+
+    public boolean add(final QueryError e) {
```

Comment:
why does this return anything?

Reply:
it doesn't need to, just because the queue interface does. I'll make void